stdlib/LinearAlgebra/src/symmetriceigen.jl

```diff
@@ -184,6 +184,65 @@ function eigen!(A::AbstractMatrix, C::Cholesky; sortby::Union{Function,Nothing}=
     GeneralizedEigen(sorteig!(vals, vecs, sortby)...)
 end
 
+# Bunch-Kaufmann (LDLT) based solution for generalized eigenvalues and eigenvectors
+function eigen(A::StridedMatrix{T}, B::BunchKaufman{T,<:AbstractMatrix}; sortby::Union{Function,Nothing}=nothing) where {T<:BlasFloat}
+    eigen!(copy(A), copy(B); sortby)
+end
+function eigen!(A::StridedMatrix{T}, B::BunchKaufman{T,<:StridedMatrix}; sortby::Union{Function,Nothing}=nothing) where {T<:BlasFloat}
+   # NOTE: Copy of BunchKaufman's getproperty(B, :D) and getproperty(B, :L/U) in bunchkaufman.jl as 'getproperty' is not in place.
+   if B.rook
+        LUD, od = LAPACK.syconvf_rook!(B.uplo, 'C', B.LD, B.ipiv)
+    else
+        LUD, od = LAPACK.syconv!(B.uplo, B.LD, B.ipiv)
+    end
+    if B.uplo == 'U'
+        M = UnitUpperTriangular(LUD)
+        du = od[2:end]
+        # Aliasing of dl and du is not allowed for lu!(Tridiagonal(dl, diag(LUD), du) below.
+        dl = B.symmetric ? copy(du) : conj.(du)
+    else
+        M = UnitLowerTriangular(LUD)
+        dl = od[1:end-1]
+        # Aliasing of dl and du is not allowed for lu!(Tridiagonal(dl, diag(LUD), du) below.
+        du = B.symmetric ? copy(dl) : conj.(dl)
+    end
```

Comment:
How about moving this to `bunchkaufman.jl`, and call it `_getproperties`, or `bk` inserted somewhere? Perhaps it's gonna be a slightly type-unstable function, but so is this code in here. Also, it could be reused then in cases where all properties of `BunchKaufman` are needed.

Reply:
Thank you for making the code generic! In fact, the above is a sore spot that I wanted to also discuss with you. The current `B.D` and `B.L` call the `getproperty` function twice. The above function returns both `B.D` and `B.L` in a single call, and also reuses the memory in `LUD`. Is there any way we can do this neatly?

Reply:
That's what I mean: make a function `getproperties(B::BunchKaufman) -> (B.D, B.L, B.p)` with exactly the code that you have, i.e., call `syconv*!` only once and return "all" useful properties at once: the `Tridiagonal` (with unaliased off-diagonals), the `Unit*Triangular`, and the permutation. As it stands, you have this code now twice, which calls for hoisting it out. I imagine that this is useful to any user who wants to write performance-optimized code involving `BunchKaufman`. We should then document that function and announce it in the NEWS, as we should announce the general new feature here, actually.

Reply:
Perfect! Sounds good...

Reply:
The method signature should just be tight enough that it catches the LAPACK cases and leaves room for another, generic method in that other BK-PR (not sure though if B.L, B.D "suffers" from the same double function call issue, but if yes, this would be even more helpful).